BugsnagPerformance/Assets/BugsnagPerformance/Scripts/Public/BugsnagPerformance.cs

```diff
@@ -332,15 +334,55 @@ private void OnRequestComplete(BugsnagUnityWebRequest request)
 
         private void EndNetworkSpan(BugsnagUnityWebRequest request, bool abort = false)
         {
-            var discard = abort || request.isHttpError || request.isNetworkError;
+            if (request == null)
+            {
+                return;
+            }
             lock (_networkSpansLock)
             {
-                if (!discard && _networkSpans.ContainsKey(request))
+                var networkSpanKey = _networkSpans.Keys.FirstOrDefault(key =>
                 {
-                    var span = _networkSpans[request];
-                    span.EndNetworkSpan(request);
+                    if (key.TryGetTarget(out var targetRequest))
+                    {
+                        return targetRequest == request;
+                    }
+                    return false;
+                });
+
+                if (networkSpanKey != null && _networkSpans.TryGetValue(networkSpanKey, out var span))
+                {
+                    if (abort || request.isHttpError || request.isNetworkError)
+                    {
+                        span.Discard();
+                    }
+                    else
+                    {
+                        span.EndNetworkSpan(request);
+                    }
+                    _networkSpans.Remove(networkSpanKey);
+                }
+            }
+            CleanUpCollectedRequests();
+        }
+
+        private void CleanUpCollectedRequests()
```

Comment:
Is it worth moving this to `AppBackgrounded` so it's running periodically but not on every network span closure?

Reply:
done 👍